mmv1/third_party/terraform/.teamcity/components/projects/feature_branches/FEATURE-BRANCH-ephemeral-write-only.kt

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+// This file is maintained in the GoogleCloudPlatform/magic-modules repository and copied into the downstream provider repositories. Any changes to this file in the downstream will be overwritten.
+
+package projects.feature_branches
+
+import ProviderNameBeta
+import ProviderNameGa
+import SharedResourceNameBeta
+import SharedResourceNameGa
+import SharedResourceNameVcr
+import builds.*
+import generated.ServicesListBeta
+import generated.ServicesListGa
+import jetbrains.buildServer.configs.kotlin.Project
+import replaceCharsId
+import vcs_roots.HashiCorpVCSRootBeta
+import vcs_roots.HashiCorpVCSRootGa
+import vcs_roots.ModularMagicianVCSRootBeta
+import vcs_roots.ModularMagicianVCSRootGa
+import getServicesList
+
+const val featureBranchEphemeralWriteOnly = "FEATURE-BRANCH-ephemeral-write-only"
+const val EphemeralWriteOnlyTfCoreVersion = "1.11.0-alpha20241211"
+
+fun featureBranchEphemeralWriteOnlySubProject(allConfig: AllContextParameters): Project {
+
+    val trigger  = NightlyTriggerConfiguration(
+        branch = "refs/heads/$featureBranchEphemeralWriteOnly" // triggered builds must test the feature branch
+    )
+    val vcrConfig = getVcrAcceptanceTestConfig(allConfig) // Reused below for both MM testing build configs
+
+    // GA
+    val gaConfig = getGaAcceptanceTestConfig(allConfig)
+    // These are the packages that have resources that will use write-only attributes
+    var ServicesListWriteOnlyGA = getServicesList(arrayOf("compute", "secretmanager", "sql", "bigquerydatatransfer"), "GA")
```

Comment:
Good to see that the new project will be limited to only the necessary services! 🙌🏻 Though the Compute build is one of our longest running builds... so running those tests in the feature branch testing project as well as the nightlies could cause some issues.

That's something to keep an eye on, at least. You could even add services to this list progressively over time, instead of adding them all now, to limit impact. Or you could change the trigger time (see other comment)

Reply:
There was also the idea of changing the test prefix within the write-only subproject to only run Tests with `TestAccEphemeral_` though we would miss out on making sure all other tests aren't affected.

I could ensure that all other tests pass by manually triggering them to run the suite entirely, I'm leaning towards this in order to not have to make any TC updates in the future such as including it as part of the `getServices` list.

Reply:
That sounds good to me, as long as you're aware there's a potential problem and you've got options about how to navigate it then I'm good 👍🏻